qa/tasks/ceph_test_case.py

```diff
@@ -134,9 +138,12 @@ def __exit__(self, exc_type, exc_val, exc_tb):
                 except CommandFailedError:
                     pass
 
-                if not self.match():
+                if emit and not self.match():
                     log.error("Log output: \n{0}\n".format(self.watcher_process.stdout.getvalue()))
                     raise AssertionError("Expected log message not found: '{0}'".format(expected_pattern))
+                elif fail or (not emit and self.match()):
+                    log.error("Log output: \n{0}\n".format(self.watcher_process.stdout.getvalue()))
+                    raise AssertionError("Unexpected log message found: '{0}'".format(expected_pattern))
```

Comment:
I'm a bit unclear about this. If we dont want to `emit` and if we found the match, so this code allows us to check if there is a string that we want to avoid but was found in cluster logs? If my rationale is true, maybe we can have better name than `emit`? Cause reading first time `emit` makes me think 'emitting output to logs' or similar

Reply:
I'm open to suggestions.

Reply:
something like `expected_in_log`?

Reply:
`is_expected` sounds clear to me. WDYT?

Reply:
`if expected_in_log and self.match()`, `elif not expected_in_log and self.match()`, `elif fail or (not expected_in_log and self.match())`

`if is_expected and self.match()`, `elif not is_expected and self.match()`, `elif fail or (not is_expected and self.match())`

`is_expected` doesn't match up with these statements that `expected`, `expected_in_log`, etc would. Although I'm fine with it but I'd vote for names starting with `expected`

Reply:
How about `present`? (see latest commits)

Reply:
yeah, `present` works. Only change I think needs to be done is use of f-string

Reply:
done